eventhubs-streamanalytics-eventhubs/tools/eh-asa-perfmon/EventReceiver.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace StreamingAtScale
+{
+    public class EventReceiver
+    {
+        private readonly object _lock = new object();
+
+        public async Task Receive(string connectionString, CancellationToken cancellationToken)
+        {
+            // In here, our consumer will read from the latest position instead of the earliest.  As a result, it won't see events that
+            // have previously been published. 
+            //
+            // Each partition of an Event Hub represents potentially infinite stream of events.  When a consumer is reading, there is no definitive
+            // point where it can assess that all events have been read and no more will be available.  As a result, when the consumer reaches the end of
+            // the available events for a partition, it will continue to wait for new events to arrive so that it can surface them to be processed. During this
+            // time, the iterator will block.
+            //
+            // In order to prevent the consumer from waiting forever for events, and blocking other code, there is a method available for developers to
+            // control this behavior. It's signaling the cancellation token passed when reading will cause the consumer to stop waiting and end iteration
+            // immediately.  This is desirable when you have decided that you are done reading and do not wish to continue.  
+
+            await using (var consumerClient = new EventHubConsumerClient(EventHubConsumerClient.DefaultConsumerGroupName, connectionString))
+            {
+                Console.WriteLine("The application will now start to listen for incoming message.");
+
+                // Each time the consumer looks to read events, we'll ask that it waits only a short time before emitting
+                // an empty event, so that our code has the chance to run without indefinite blocking.
+
+                ReadEventOptions readOptions = new ReadEventOptions
+                {
+                    MaximumWaitTime = TimeSpan.FromMilliseconds(500)
+                };
+
+                using (var csvOutput = File.CreateText("./result.csv"))
+                {
+                    csvOutput.WriteLine("EventCount,BatchCount,BatchFrom,BatchTo,MinLatency,MaxLatency,AvgLatency");
+                    try
+                    {
+                        // The client is safe and intend to be long-lived; in order to begin reading at the
+                        // latest event, we'll need to specify that reading should not start at earliest.
+
+                        var events = consumerClient.ReadEventsAsync(startReadingAtEarliestEvent: false, readOptions, cancellationToken);
+
+                        await foreach (PartitionEvent currentEvent in events)
```

Comment:
I'd recommend merging lines 51 and 53.  The typical pattern in C#, and the one that we'd like to represent here should be:

```csharp
await foreach (PartitionEvent currentEvent in consumerClient.ReadEventsAsync(readOptions, cancellationToken))
{
    // << more code >>
}
```

Reply:
Resolved